packages/slope/src/adapter.ts

```diff
@@ -0,0 +1,191 @@
+import {
+    BaseSignerWalletAdapter,
+    pollUntilReady,
+    WalletAccountError,
+    WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58';
+
+interface SlopeWallet {
+    connect(): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+        };
+    }>;
+    disconnect(): Promise<{
+        msg: string;
+    }>;
+    signTransaction(message: string): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signature: string;
+        }
+    }>;
+    signAllTransactions(messages: string[]): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signatures: string[];
+        }
+    }>;
+}
+
+interface SlopeWindow extends Window {
+    Slope?: any;
+}
+
+declare const window: SlopeWindow;
+
+export interface SlopeWalletAdapterConfig {
+    pollInterval?: number;
+    pollCount?: number;
+}
+
+export class SlopeWalletAdapter extends BaseSignerWalletAdapter {
+    private _connecting: boolean;
+    private _wallet: SlopeWallet | null;
+    private _publicKey: PublicKey | null;
+
+    constructor(config: SlopeWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (!this.ready) pollUntilReady(this, config.pollInterval || 1000, config.pollCount || 3);
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get ready(): boolean {
+        return !!window.Slope;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._publicKey;
+    }
+
+    get autoApprove(): boolean {
+        return false;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            this._connecting = true;
+
+            if (!window.Slope) throw new WalletNotFoundError();
+
+            const wallet = this._wallet || new window.Slope();
+            if (!wallet.connect) throw new WalletNotInstalledError()
+
+            let account: string;
+            try {
+               const { msg, data } = await wallet.connect();
+
+               if (msg !== 'ok') {
+                   throw new WalletAccountError(msg);
+               }
+
+               account = data.publicKey;
+            } catch (error) {
+                throw new WalletAccountError(error?.message, error);
+            }
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(account);
+            } catch (error) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            wallet.isConnected = true;
```

Comment:
`wallet` doesn't define (in the TypeScript above, at least) an `isConnected` property, so this should fail TS compilation. What's the reason to set `isConnected` on the Slope object? Wouldn't Slope already know whether it's connected or not?

Reply:
Removed